scanpy/plotting/_tools/scatterplots.py

```diff
@@ -205,8 +237,44 @@ def plot_scatter(
                 **kwargs,
             )
         else:
+            if add_contour:
+                # the default contour is a black edge followd by a
+                # thin white edged added around connected clusters.
+                # To add a contour
+                # three overlapping scatter plots are drawn:
+                # First black dots with slightly larger size,
+                # then, white dots a bit smaller, but still larger
+                # than the final dots. Then the final dots are drawn
+                # with some transparency.
+
+                black_width, white_width = contour_config['edge_width']
```

Comment:
Please rename `black_*` to `outline_*` and `white_*` to `bg_*` or something similar. For bright-on-dark plots the non-semantic names you chose don’t make sense :wink: